hooks/useLPActivityAndImplicitSwaps/useLPActivityAndImplicitSwaps.ts

```diff
@@ -0,0 +1,202 @@
+import { CurrencyAmount, Price } from '@uniswap/sdk-core';
+import { priceToClosestTick, TickMath } from '@uniswap/v3-sdk';
+import { ethers } from 'ethers';
+import { ActivityType } from 'hooks/useActivity/useActivity';
+import { usePoolTokens } from 'hooks/usePoolTokens';
+import { useSlot0 } from 'hooks/useSlot0';
+import { checkZero } from 'hooks/useSwapPositionsData/helpers';
+import {
+  getAmount0FromLPStats,
+  getAmount1FromLPStats,
+} from 'lib/controllers/uniswap';
+import { uniTokenToErc20Token } from 'lib/uniswapSubgraph';
+import { useCallback, useMemo } from 'react';
+import {
+  LpActivityByAccountDocument,
+  LpActivityByAccountQuery,
+} from 'types/generated/graphql/inKindSubgraph';
+import { useQuery } from 'urql';
+
+export function useLPActivityAndImplicitSwaps(
```

Comment:
I don't love this API. I'm not sure how much to design around my table use case, but there it would be nice to just be able to pass an activity and get the implicit swaps. But we need to know the the tick of the previous LP event... So we could change our subgraph to try to keep track of this? Perhaps we mutate the position entity with the lastKnownTick? Feels like it would be best to get all the info into a single activity entity.